lexicon/conversion.py

```diff
@@ -140,6 +140,69 @@ def run(self):
             lexicon_file.write(ET.tostring(root, "unicode"))
 
 
+class LexiconFromTextFileJob(Job):
+    """
+    Create a bliss lexicon from a regular text file, where each line contains:
+    < WORD > < PHONEME1 > < PHONEME2 > ...
+    separated by tabs or spaces.
+    The lemmata will be added in the order they appear in the text file,
+    the phonemes will be sorted alphabetically.
+    No special lemmata or phonemes are added.
+
+    As the splitting is taken from RASR and not fully tested,
+    it might not work in all cases so do not use this job
+    without checking the output manually on new lexica.
+    """
+
+    def __init__(self, text_file, compressed=True):
+        """
+        :param Path text_file:
+        :param compressed: save as .xml.gz
+        """
+        self.text_file = text_file
+
+        self.out_bliss_lexicon = self.output_path(
+            "lexicon.xml.gz" if compressed else "lexicon.xml"
+        )
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        lex = lexicon.Lexicon()
+
+        phonemes = set()
+        last_lemma = None
+        with uopen(self.text_file.get_path()) as f:
+            for line in f:
+                # splitting is taken from RASR
+                # src/Tools/Bliss/blissLexiconLib.py#L185
+                s = line.split(None, 1)
+                orth = s[0].split("\\", 1)[0]
+                phon_variants = [
+                    tuple(p.split()) for p in s[1].split("\\") if p.strip()
+                ]
+                for phon_variant in phon_variants:
+                    for phon in phon_variant:
+                        if phon not in phonemes:
+                            phonemes.add(phon)
+                phon = [" ".join(v) for v in phon_variants]
+                lemma = lexicon.Lemma(orth=[orth], phon=phon)
+                if last_lemma and lemma.orth[0] == last_lemma.orth[0]:
+                    last_lemma.phon += phon
+                else:
+                    lex.add_lemma(lemma)
+                    last_lemma = lemma
+
+        for phoneme in sorted(phonemes):
+            lex.add_phoneme(phoneme)
+
+        with uopen(self.out_bliss_lexicon.get_path(), "wb") as lexicon_file:
+            s = ET.tostring(lex.to_xml(), "unicode")
+            pretty_s = minidom.parseString(s).toprettyxml(indent=" ", encoding="utf-8")
```

Comment:
We have a function in `util` which you might consider for writing the xml file.

https://github.com/rwth-i6/i6_core/blob/main/util.py#L204